basiclti/basiclti-impl/src/java/org/sakaiproject/lti/impl/LTIExportServiceImpl.java

```diff
@@ -0,0 +1,254 @@
+		}
+		if (!"!admin".equals(siteId)) return false;
+		return siteService.allowUpdateSite(siteId);
+	}
+
+	/* (non-Javadoc)
+	 * @see impl.LTIExportService#export(java.io.OutputStream, int, java.lang.String)
+	 */
+	public void export(OutputStream out, String siteId, int exportType, String filterId) {
+		String search = null;
+		//check if we need to filter the tools by tool_id
+		if(StringUtils.isNotEmpty(filterId)) {
+			search = "tool_id:"+filterId;
+		}
+		List<Map<String,Object>> contents = ltiService.getContentsDao(search, null, 0, 0, siteId, isAdmin(siteId));
+		switch(exportType) {
+			case TYPE_CSV : exportCSV(out, siteId, contents); break;
+			case TYPE_EXCEL : exportExcel(out, siteId, contents); break;
+		}
+	}
+	
+	private String getCSVSeparator() {
+		if(csvSeparator == null) {
+			csvSeparator = serverConfigurationService.getString("basiclti.export.csv.separator", ",");
+		}
+		return csvSeparator;
+	}
+	
+	private void exportCSV(OutputStream out, String siteId, List<Map<String,Object>> contents) {
+		
+		StringBuilder result = new StringBuilder();
+		FormattedText validator = new FormattedText();
+		DateFormat dateFormatter = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.SHORT, rb.getLocale());
+		
+		String attribution_name = serverConfigurationService.getString(LTIService.LTI_SITE_ATTRIBUTION_PROPERTY_NAME);
+		if(StringUtils.isNotEmpty(attribution_name)) {
+			//check if given property is a translation key
+			String aux = rb.getString(attribution_name);
+			if(StringUtils.isNotEmpty(aux)) 
+				attribution_name = aux;
+		}
+		
+		boolean isAdmin = isAdmin(siteId);
+		
+		//set header row
+		result.append(rb.getString("export.title"));
+		result.append(getCSVSeparator());
+		result.append(rb.getString("export.url"));
+		result.append(getCSVSeparator());
+		if(isAdmin) {
+			result.append(rb.getString("export.siteid"));
+			result.append(getCSVSeparator());
+			result.append(rb.getString("export.sitetitle"));
+			result.append(getCSVSeparator());
+		}
+		result.append(rb.getString("export.createdat"));
+		if(isAdmin) {
+			result.append(getCSVSeparator());
+			result.append(rb.getString("export.sitecontactname"));
+			result.append(getCSVSeparator());
+			result.append(rb.getString("export.sitecontactemail"));
+			if(StringUtils.isNotEmpty(attribution_name)) {
+				result.append(getCSVSeparator());
+				result.append(attribution_name);
+			}
+		}
+		result.append("\n");
+		
+		//values rows
+		for(Map<String,Object> content : contents) {
+			result.append(validator.escapeHtml((String)content.get("title"), false));
+			result.append(getCSVSeparator());
+			
+			String url = (String)content.get("launch");
+			if(StringUtils.isEmpty(url)) {
+				try {
+					url = (String)(ltiService.getToolDao(new Long(content.get(LTIService.LTI_TOOL_ID).toString()), siteId).get("launch"));
+				} catch(Exception e) {
+					url = "-";
+					e.printStackTrace();
```

Comment:
Shouldn't have `e.printStackTrace()` use logging instead, although this seems to be expected so does it need logging at all?